src/server/pps/server/api_server.go

```diff
@@ -1071,29 +1073,69 @@ func (a *apiServer) RestartDatum(ctx context.Context, request *pps.RestartDatumR
 // listDatum contains our internal implementation of ListDatum, which is shared
 // between ListDatum and ListDatumStream. When ListDatum is removed, this should
 // be inlined into ListDatumStream
-func (a *apiServer) listDatum(pachClient *client.APIClient, job *pps.Job, page, pageSize int64) (response *pps.ListDatumResponse, retErr error) {
+func (a *apiServer) listDatum(pachClient *client.APIClient, job *pps.Job, input *pps.Input, page, pageSize int64) (response *pps.ListDatumResponse, retErr error) {
 	if _, err := checkLoggedIn(pachClient); err != nil {
 		return nil, err
 	}
 	response = &pps.ListDatumResponse{}
 	ctx := pachClient.Ctx()
 	pfsClient := pachClient.PfsAPIClient
+	if job == nil && input == nil {
+		return nil, errors.Errorf("must specify either job or input to list datums from")
+	}
+	if job != nil && input != nil {
+		return nil, errors.Errorf("can't specify both job and input to list datums from")
+	}
 
-	// get information about 'job'
-	jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
-		Job: &pps.Job{
-			ID: job.ID,
-		},
-	})
-	if err != nil {
-		return nil, err
+	var pipelineName string
+	var statsCommit *pfs.Commit
+	var ji *pps.JobInfo
+
+	if job != nil {
+		// get information about 'job'
+		jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
+			Job: &pps.Job{
+				ID: job.ID,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		input = jobInfo.Input
+		pipelineName = jobInfo.Pipeline.Name
+		statsCommit = jobInfo.StatsCommit
+		ji = jobInfo
+	}
+	if input != nil {
```

Comment:
should this be an else if? The job if above sets input, and I assume we still want to allow listDatum for existing jobs with cron inputs